asserts/asserts.go

```diff
@@ -1021,6 +1052,12 @@ func assemble(headers map[string]interface{}, body, content, signature []byte) (
 		return nil, fmt.Errorf("unknown assertion type: %q", typ)
 	}
 
+	if assertType.flags&jsonBody != 0 {
+		if err := checkJSON(assertType, body); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
for consistency don't we want this closer to the utf8.Valid  check?

Reply:
it needs the assertType which we get and check after. The intention here was also to have it next to the other flag checks